fog/ingest/server/src/controller.rs

```diff
@@ -782,8 +830,16 @@ where
         log::info!(self.logger, "Key successfully set in enclave: {}", pubkey);
 
         *self.last_sealed_key.lock().unwrap() = None;
-        self.write_state_file();
-        Ok(self.get_ingest_summary())
+        self.write_state_file_inner(&mut state);
+        let result = self.get_ingest_summary_inner(&mut state);
+
+        // Don't hold the state mutex while we are talking to IAS
```

Comment:
i don't know that this is really critical, but it seems like a good idea and can't hurt